docs/docs/workers.md

```diff
@@ -39,6 +39,9 @@ By default, workers will start working immediately and will block and wait for
 new work when they run out of work.  Workers can also be started in _burst
 mode_ to finish all currently available work and quit as soon as all given
 queues are emptied.
+In _burst mode_, a worker will also quit if there are no available jobs
+prepared to start, even when there exists a non-empty queue. For example, if
```

Comment:
I'm ok with adding a note to clarify that deferred jobs will not cause workers in burst mode to block. But the statement is a bit misleading because jobs that are deferred are actually not yet `enqueued`, they are actually stored in `DeferredJobRegistry`.

Reply:
> jobs that are deferred are actually not yet enqueued, they are actually stored in DeferredJobRegistry.

Yes, this is the actual behavior.
I think one current misleading part is when users execute/write the code `queue.enqueue(<func_name>, args, ..., depends_on=<another_job_id>`, it is likely to assume this job is already "enqueued" in the queue and the queue should not be empty until all jobs are finished.

One scenario is that all jobs are "enqueued" by the statement `queue.enqueue()`, and then several workers start in burst mode. It is reasonable to assume the workers will finish all jobs, including those deferred ones, before they exit. However, if at first many jobs are deferred due to dependency and some workers take over all available jobs (maybe small amount at first), then the other workers will immediately exit, which is unexpected.

Reply:
> I think one current misleading part is when users execute/write the code queue.enqueue(<func_name>, args, ..., depends_on=<another_job_id>, it is likely to assume this job is already "enqueued" in the queue and the queue should not be empty until all jobs are finished.

It's a reasonable assumption. I think this is the part that we need to clarify, not that `a worker will also quit if there are no available jobs prepared to start, even when there exists a non-empty queue` because this is technically not true.

Thanks for pointing this out.

Reply:
Right. If the queue is not empty, the worker will not quit in burst mode.
In detail, all deferred jobs are located in their corresponded registries first before getting into the queue.